controllers/configurationpolicy_controller.go

```diff
@@ -1045,277 +1216,255 @@ func (r *ConfigurationPolicyReconciler) handleDeletion(plc *policyv1.Configurati
 
-	_, _, err := unstructured.UnstructuredJSONScheme.Decode(objectT.ObjectDefinition.Raw, nil, &desiredObj)
-	if err != nil {
-		log.Error(err, "Could not decode the objectDefinition", "index", index)
+			desiredObjects = append(desiredObjects, desiredObj)
 
-		errEvent = &objectTmplEvalEvent{
-			compliant: false,
-			reason:    "K8s decode object definition error",
-			message: fmt.Sprintf("Decoding error, please check your policy file!"+
-				" Aborting handling the object template at index [%v] in policy `%v` with error = `%v`",
-				index, plc.Name, err),
+			continue
 		}
-	}
 
-	// strings.TrimSpace() is needed here because a multi-line value will have
-	// '\n' in it. This is kept for backwards compatibility.
-	desiredObj.SetName(strings.TrimSpace(desiredObj.GetName()))
-	desiredObj.SetNamespace(strings.TrimSpace(desiredObj.GetNamespace()))
-	desiredObj.SetKind(strings.TrimSpace(desiredObj.GetKind()))
+		desiredObj = &unstructured.Unstructured{}
 
-	// map raw object to a resource, generate a violation if resource cannot be found
-	if errEvent == nil {
-		scopedGVR, err = r.getMapping(desiredObj.GroupVersionKind(), plc, index)
+		_, _, err = unstructured.UnstructuredJSONScheme.Decode(rawDesiredObject, nil, desiredObj)
 		if err != nil {
-			mappingErr = err
+			log.Error(err, "Could not decode the objectDefinition", "index", index)
 
-			errEvent = &objectTmplEvalEvent{
+			errEvent := &objectTmplEvalEvent{
 				compliant: false,
-				reason:    "K8s error",
-				message:   err.Error(),
+				reason:    "K8s decode object definition error",
+				message: fmt.Sprintf("Decoding error, please check your policy file!"+
+					" Aborting handling the object template at index [%v] in policy `%v` with error = `%v`",
+					index, plc.Name, err),
 			}
-		}
-	}
-
-	// Fetch and filter namespaces using provided namespaceSelector
-	desiredNs := desiredObj.GetNamespace()
 
-	if scopedGVR.Namespaced && desiredNs == "" {
-		nsSelector := plc.Spec.NamespaceSelector
-
-		selectedNamespaces, err := r.SelectorReconciler.Get(plc.Namespace, plc.Name, nsSelector)
-		if err != nil {
-			log.Error(err, "Failed to select the namespaces",
-				"namespaceSelector", nsSelector.String())
+			return nil, &scopedGVR, errEvent, err
+		}
 
-			msg := fmt.Sprintf("Error filtering namespaces with provided namespaceSelector: %v", err)
+		// strings.TrimSpace() is needed here because a multi-line value will have
+		// '\n' in it. This is kept for backwards compatibility.
+		desiredObj.SetName(strings.TrimSpace(desiredObj.GetName()))
+		desiredObj.SetKind(strings.TrimSpace(desiredObj.GetKind()))
+		desiredObj.SetNamespace(strings.TrimSpace(desiredObj.GetNamespace()))
 
-			// only report this error if there wasn't another yet
-			if errEvent == nil {
-				errEvent = &objectTmplEvalEvent{
-					compliant: false,
-					reason:    "namespaceSelector error",
-					message:   msg,
-				}
+		if desiredObj.GetNamespace() != desiredNs && desiredObj.GetNamespace() != ns {
```

Comment:
Making sure I'm understanding this conditional correctly: this is ensuring that the resolved namespace is either an empty string (or a different marker if we so choose for `desiredNs`) or the namespace selected from the `namespaceSelector`?

EDIT: Actually, is there logic somewhere that allows skipping the empty stringed namespace?

Reply:
Oops--ignore that edit. Got my wires crossed with the ObjectName variable 🤦‍♂️